Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -1097,6 +1097,135 @@ public enum HTTPResponseStatus {
             return true
         }
     }
+
+    /// Initialize a `HTTPResponseStatus` from a given status and reason.
+    ///
+    /// - Parameter statusCode: The integer value of the HTTP response status code
+    /// - Parameter reasonPhrase: The textual reason phrase from the response.
```

Comment:
Can we document that this will be discarded in favour of the default if it matches one we know?

Reply:
@Lukasa +1... Kind of "odd" in terms of contract tho...

Reply:
@normanmaurer I agree, it feels a bit odd. Should it become an optional initializer instead such that the use case becomes as follows?

```
let status: HTTPResponseStatus = .init(statusCode: myCode) ?? .custom(code: myCode, reasonPhrase: myReason)
```

I just noticed that the associated value for the code in the `custom` case is a UInt, and not an Int as is now the case in the `init` method I added. Should that be aligned as well?

Reply:
We should have this be `UInt` for now: there's no point shipping a breaking change to move that from `UInt` to `Int`. :wink:

As to the name, I kinda feel like this type doesn't handle custom reason phrases for normal codes very well. Not sure what we can do about that really, other than bring an optional string reason phrase along with us at all times (ew), so I'm inclined to go with our current behaviour for now.

Reply:
@Lukasa huh... it was internal before so I don't think it would be breaking. What I am missing (and its possible I do ;)) ?